openedx/adg/lms/applications/test/test_forms.py

```diff
@@ -0,0 +1,117 @@
+                                       organization='test', linkedin_url=''):
+        """
+        Initialize the data dictionary for contact information forms
+        """
+        return {
+            'gender': gender,
+            'phone_number': phone_number,
+            'birth_day': birth_day,
+            'birth_month': birth_month,
+            'birth_year': birth_year,
+            'organization': organization,
+            'linkedin_url': linkedin_url
+        }
+
+    def setUp(self):
+        """
+        Setup a new user to POST the data
+        """
+        self.user = get_user_model().objects.create_user(
+            email='testuser@test.com',
+            password='password',
+            username='testuser'
+        )
+
+    def test_contact_info_form_with_future_birth_date(self):
+        """
+        Verify that future dates are not allowed in birth_date field
+        """
+        tomorrow = date.today() + timedelta(days=1)
+        data = self.contact_information_dictionary(birth_day=str(tomorrow.day),
+                                                   birth_month=str(tomorrow.month), birth_year=str(tomorrow.year))
+        form = ContactInformationForm(data=data)
+        self.assertFalse(form.is_valid())
+
+    def test_contact_info_form_with_minimum_valid_age(self):
+        """
+        Verify that birth_date with at least 21 year difference is allowed
+        """
+        years = 21
+        minimum_age = date.today() - timedelta(days=(years * DAYS_PER_YEAR + 1))
+        data = self.contact_information_dictionary(birth_day=str(minimum_age.day),
+                                                   birth_month=str(minimum_age.month), birth_year=str(minimum_age.year))
+        form = ContactInformationForm(data=data)
+        self.assertTrue(form.is_valid())
+
+    def test_contact_info_form_with_minimum_invalid_age(self):
+        """
+        Verify that birth_date with at least 21 year difference is allowed
+        """
+        years = 21
+        minimum_age = date.today() - timedelta(days=(years * DAYS_PER_YEAR))
+        data = self.contact_information_dictionary(birth_day=str(minimum_age.day),
+                                                   birth_month=str(minimum_age.month), birth_year=str(minimum_age.year))
+        form = ContactInformationForm(data=data)
+        self.assertFalse(form.is_valid())
+
+    def test_contact_info_form_with_maximum_valid_age(self):
+        """
+        Verify that birth_date with difference greater than 60 year is not allowed
+        """
+        years = 61
+        minimum_age = date.today() - timedelta(days=(years * DAYS_PER_YEAR))
+        data = self.contact_information_dictionary(birth_day=str(minimum_age.day),
+                                                   birth_month=str(minimum_age.month), birth_year=str(minimum_age.year))
+        form = ContactInformationForm(data=data)
+        self.assertTrue(form.is_valid())
```

Comment:
See if we can merge valid age tests together using `pytest.mark.parametrize` and avoid duplication.
If we can, kindly do the same for invalid tests and wherever else it is applicable.